src/methods/method_overloading/hard/q3/MainTest.java

```diff
@@ -0,0 +1,87 @@
+package methods.method_overloading.hard.q3;
+
+import java.util.stream.Stream;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import global.BaseTest;
+import global.tools.CustomAssertions;
+import global.tools.TestOption;
+import global.utils.MethodUtil;
+import global.variables.Clause;
+import global.variables.clauses.DoubleLiteral;
+import global.variables.clauses.IntegerLiteral;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+
+public class MainTest extends BaseTest {
+  // Java
+  public Clause[] testSentence() {
+    return new Clause[] {
+        new StringLiteral("The sum of the 2 numbers: "),
+        new IntegerLiteral("sumInt"),
+        new NewLine(),
+        new StringLiteral("The sum of the 2 numbers: "),
+        new DoubleLiteral("sumDouble"),
+        new NewLine()
+    };
+  }
+
+  public void runMain() {
+    Overload3.main(new String[0]);
+  }
+
+  static Stream<Arguments> integerInputProvider() {
+    return Stream.of(
+        Arguments.of(1, 2, 3),
+        Arguments.of(0, 0, 0),
+        Arguments.of(1, 0, 1),
+        Arguments.of(1, -1, 0),
+        Arguments.of(-99, 201, 102));
+  }
+
+  static Stream<Arguments> doubleInputProvider() {
+    return Stream.of(
+        Arguments.of(1.1, 2.5, 3.6),
+        Arguments.of(0.01, 0.01, 0.02),
+        Arguments.of(-1.0, 1.03, 0.03),
+        Arguments.of(1.5, -1.5, 0.0),
```

Comment:
There should also be a case where each is an integer because that should also work